bukkit/src/main/java/dev/aurelium/auraskills/bukkit/requirement/blocks/RequirementNode.java

```diff
@@ -0,0 +1,16 @@
+package dev.aurelium.auraskills.bukkit.requirement.blocks;
+
+import dev.aurelium.auraskills.bukkit.AuraSkills;
+import org.bukkit.entity.Player;
+
+public abstract class RequirementNode {
+
+    protected AuraSkills plugin;
+
+    public RequirementNode(AuraSkills plugin) {
```

Comment:
The deny message can be stored and passed in the constructor of the abstract class here. Then you only need to implement the getDenyMessage method once here and not multiple times in each requirement type.